pkg/cloudprovider/plugins/azure/azure_compute.go

```diff
@@ -113,32 +133,41 @@ func (computeCfg *computeServiceConfig) waitForInventoryInit(duration time.Durat
 }
 
 // getCachedVirtualMachines returns virtualMachines from the cache for the subscription.
-func (computeCfg *computeServiceConfig) getCachedVirtualMachines() []*virtualMachineTable {
-	snapshot := computeCfg.resourcesCache.GetSnapshot()
-	if snapshot == nil {
-		azurePluginLogger().V(4).Info("Compute service cache snapshot nil", "type", providerType, "account", computeCfg.account)
+func (computeCfg *computeServiceConfig) getCachedVirtualMachines(selector *types.NamespacedName) []*virtualMachineTable {
```

Comment:
All VMs, regardless of selector store

Reply:
added a new function for fetching all cached vms.